src/Operations.jl

```diff
@@ -1702,7 +1702,8 @@ end
 
 function is_package_downloaded(project_file::String, pkg::PackageSpec)
     sourcepath = source_path(project_file, pkg)
-    @assert sourcepath !== nothing
+    identifier = pkg.name !== nothing ? pkg.name : pkg.uuid
+    (sourcepath === nothing) && pkgerror("Could not locate the source code for the $(identifier) package. Are you trying to use a manifest generated by a different version of Julia?")
```

Comment:
```suggestion
    (sourcepath === nothing) && pkgerror("Could not locate the source code for the $(identifier) package. This can happen if the manifest was generated by another julia version. Run `import Pkg; Pkg.resolve()` to update the manifest")
```
Perhaps give actionable advice? Worth double-checking the advice to re-resolve fixes it though

Reply:
Does this step actually work? I was under the impression that it will still error.

Reply:
I'm going to merge this PR now, but if we can confirm that this advice actually does work, then we can edit the error message and add this advice.